sda-commons-asyncapi/src/main/java/org/sdase/commons/spring/boot/asyncapi/jsonschema/JsonSchemaBuilder.java

```diff
@@ -77,6 +78,9 @@ default Map<String, JsonNode> toJsonSchema(Iterable<Type> types) {
                 Map.Entry::getKey,
                 Map.Entry::getValue,
                 (v1, v2) -> {
+                  if (Objects.equals(v1, v2)) {
```

Comment:
We can also use .distinct() in the stream e.g:

 ```java
 return StreamSupport.stream(types.spliterator(), false)
        .distinct()
        .map(this::toJsonSchema)
        .map(Map::entrySet)
        .flatMap(Set::stream)
        .distinct()
        .collect(...
```

Reply:
Changed the implementation and added a comment.